Code/Framework/AzFramework/AzFramework/Entity/GameEntityContextComponent.h

```diff
@@ -74,6 +79,49 @@ namespace AzFramework
+        // Expanded Entity State Handling to Introduce Hierarchichal Entity Activation Handling
+
+        //! Utility method that parses the 'childrenByParentTree' in order to return a hierarchichal list of entities from top to bottom.
+        //! \param entityId The root entity ID to gather the tree from.
+        void AddEntityToParentChildTree(AZ::Entity* entity);
+        //! Utility method that parses the 'childrenByParentTree' in order to return a hierarchichal list of entities from top to bottom.
+        //! \param entityId The root entity ID to gather the tree from.
+        void RemoveEntityFromParentChildTree(AZ::Entity* entity);
+        //! Utility method that parses the 'childrenByParentTree' in order to return a hierarchichal list of entities from top to bottom.
+        //! \param entityId The root entity ID to gather the tree from.
+        void RemoveEntityFromParentChildTreeById(const AZ::EntityId& entityId);
+        //! Utility method that parses the 'childrenByParentTree' in order to return a hierarchichal list of entities from top to bottom.
+        //! \param entityId The root entity ID to gather the tree from.
+        //! \param entityId The root entity ID to gather the tree from.
+        void UpdateParentChildMaps(AZ::EntityId child, AZ::EntityId oldParent, AZ::EntityId newParent);
+        //! Utility method that parses the 'childrenByParentTree' in order to return a hierarchichal list of entities from top to bottom.
+        //! \param entityId The root entity ID to gather the tree from.
+        void RecomputeEffectiveActivationForEntity(AZ::Entity* movedChildEntity);
+
+        //! Utility method that parses the 'childrenByParentTree' in order to return a hierarchichal list of entities from top to bottom.
+        //! \param entityId The root entity ID to gather the tree from.
+        //! \param entityId The root entity ID to gather the tree from.
+        void GetEntityTreeFromRootEntity(const AZ::EntityId& entityId, AZStd::vector<AZ::EntityId>& out);
+        //! Utility method that parses the 'childrenByParentTree' in order to return a hierarchichal list of entities from top to bottom.
+        //! \param entityId The root entity ID to gather the tree from.
+        //! \param entityId The root entity ID to gather the tree from.
+        //! \param entityId The root entity ID to gather the tree from.
+        void PruneDescendantsFromTreeInPlace(const AZ::EntityId& root, AZStd::vector<AZ::EntityId>& list, size_t fromIndex) const;
+
+        //! Local stored hierarchy tree to enable hierarchy handling without the TransformBus (which is only functional while active.)
+        //! Tree is updated at
+        AZStd::unordered_map<AZ::EntityId, AZStd::vector<AZ::EntityId>, AZStd::hash<AZ::EntityId>> childrenByParentTree; // Parent -> [Children..]
+        AZStd::unordered_map<AZ::EntityId, AZ::EntityId, AZStd::hash<AZ::EntityId>> parentOf; // Child -> Parent
```

Comment:
just nit, naming, we use m_ to indicate member variable, g_ global, s_static, etc